core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassMonitor.scala

```diff
@@ -38,88 +35,66 @@ class FastPassMonitor private (dataSource: SlickDataSource,
 
   private val openTelemetryTags: Map[String, String] = Map("service" -> "FastPassMonitor")
   override def receive: Receive = { case DeleteExpiredGrants =>
-    openTelemetry.incrementCounter("fastpass-monitor-receive-msg", tags = openTelemetryTags).unsafeToFuture()
-    deleteExpiredGrants()
+    for {
+      _ <- openTelemetry.incrementCounter("fastpass-monitor-sweeper-start", tags = openTelemetryTags).unsafeToFuture()
+      _ <- deleteExpiredGrants()
+    } yield ()
   }
 
   /*
    * Remove google roles for expired grants and delete the grants from the database.
    * This is done in a transaction so that if any of the google calls fail, the grants are not deleted.
    * In order to reduce the number of api and db calls, we group the grants by workspaceId, accountEmail, and resourceType.
    */
-  private def deleteExpiredGrants(): Future[ReadWriteAction[Unit]] =
+  private def deleteExpiredGrants(): Future[Unit] =
+    for {
+      grantsGroupedByEmail <- findFastPassGrantsToRemove()
+    } yield Future.sequence(removeFastPassGrants(grantsGroupedByEmail))
+
+  private def findFastPassGrantsToRemove(): Future[Iterable[((Workspace, WorkbenchEmail), Seq[FastPassGrant])]] =
     dataSource.inTransaction { dataAccess =>
-      dataAccess.fastPassGrantQuery
-        .findExpiredFastPassGrants()
-        .flatMap { expiredGrants =>
-          logger.info(s"Found ${expiredGrants.size} total expired grants")
-          // Convert to stream and group by workspaceId
-          DBIO.sequence(expiredGrants.to(LazyList).groupBy(_.workspaceId).map { case (workspaceId, workspaceGrants) =>
-            logger.info(s"Found ${workspaceGrants.size} expired grants for workspace $workspaceId")
-            // Query workspace info then group by accountEmail
-            dataAccess.workspaceQuery.findByIdOrFail(workspaceId).flatMap { workspace =>
-              DBIO.sequence(workspaceGrants.groupBy(_.accountEmail).map { case (_, accountEmailGrants) =>
-                // Remove grants for a given workspace and accountEmail
-                removeGrantsForAccountEmailInWorkspace(dataAccess, workspace, accountEmailGrants)
-              })
-            }
-          })
+      for {
+        expiredGrants <- dataAccess.fastPassGrantQuery.findExpiredFastPassGrants()
+        _ = logger.info(s"Found ${expiredGrants.size} FastPass grants to clean up")
+        groupedByWorkspaceId = expiredGrants.groupBy(_.workspaceId)
```

Comment:
I'm not fully grokking why we need all of these `groupBys`. From what I can tell, a `FastPassGrant` has all the information Rawls needs to revoke the grant except for the workspace's Google project. Could we instead iterate over a list of `(FastPassGrant, GoogleProject)`s and revoke the IAM grant accordingly? 

Let me know if you'd like to hop on a call and discuss!

Reply:
Is it true that workspaces are 1:1 with a google project?

Reply:
ah, ok. Thanks PPW! Updated and pushed